### PR TITLE
feat: ready callback

### PR DESCRIPTION
**Pull request by tarsinzer**
It might solve https://github.com/dkaoster/scrolly-video/issues/46

Rewrote` decodeVideo` a little bit for better readability (more linear reading with early return and async-await)

Btw, you do catch first because sometimes it can just produce the empty array by default without `catch`, right?
As I understand, you put `catch` first to "normalize" an empty array and catch both cases later in the last `then`.
Because I wanted to move catch to the bottom but then I understood that it might be a normalization.

### CODE REVIEW

File: src/ScrollyVideo.js [hunk 1]
```diff
@@ -62,8 +63,8 @@ class ScrollyVideo {
     this.useWebCodecs = useWebCodecs;
     this.cover = cover;
     this.sticky = sticky;
-    this.full = full;
```
Comment (dkaoster) [inline src/ScrollyVideo.js:65]: why remove this?

**Review by dkaoster — COMMENTED** [review]
Sorry for the delay! This looks pretty good to me, just one comment and also think we will need to add the same callback into the svelte and vue versions?

**Comment by polyrhythmatic** [comment]
Anything I can do to help merge? Could use this feature right now!

File: src/ScrollyVideo.js (hunk 1 above)
Comment (tarsinzer) [reply src/ScrollyVideo.js:65]: `this.full` is not being used anywhere in this class

`full` arg in constructor is used on instantiation only:
CleanShot 2024-05-02 at 08 22 02@2x

**Comment by tarsinzer** [comment]
@dkaoster it will be applied to svelte and vue by default with other props, because of `restProps` 

svelte
```
const { videoPercentage, ...restProps } = $$props;
```

vue
```
const { videoPercentage, ...restProps } = this.$attrs;
```

So, probably we can merge it, right?

**Review by dkaoster — APPROVED** [review]

**Pull request merged**